test/common/testlib.py

```diff
@@ -2168,7 +2168,7 @@ def nondestructive(testEntity):
 
     Can be used on test classes and individual test methods.
     """
-    testEntity._testlib__nondestructive = True
+    setattr(testEntity, '_testlib__nondestructive', True)
```

Comment:
The "standard practice" here was maybe something I got confused about.  I thought I had fixed a `mypy` issue in `systemd_ctypes` by doing this after reading something online, but I can't find either of those things anymore, so maybe I was imagining something else.

The error that we fix by this is that `vulture` finds no user of this variable that we set.  The reason for that is that we access it like so:

```python
  def get_decorator(method, _class, name, default=None):
      """Get decorator value of a test method or its class
  
      Return None if the decorator was not set.
      """
      attr = "_testlib__" + name
      return getattr(method, attr, getattr(_class, attr, default))
```

So this is basically about using `setattr()` to match `getattr()`.

Maybe we want a helper like `set_decorator()` or so to do with it, but this would end up being very little more than a single `setattr()` call...

If nothing else, this commit could probably use a better commit message.

Reply:
Aah, now I at least understand where this is coming from. Explaining it in the commit message and keeping as-is WFM. Thanks!